Devex 10337 - AWS v2 -> v3 conversion: SQS

jtdevos opened the pull request:
Mostly straightforward changes to update to v3 api. However, I deleted `@web-api/persistence/sqs/getMessages.ts` because I found zero references to it and because it's implementation seems broken.

flexion#10337

jtdevos committed:
10337: first pass at v2->v3 update, need to update tests (wip)

jimlerza commented on shared/src/business/useCases/users/generateChangeOfAddress.ts:
```diff
@@ -88,33 +89,29 @@ const generateChangeOfAddressForPractitioner = async ({
+        QueueUrl: `https://sqs.${process.env.REGION}.amazonaws.com/${process.env.AWS_ACCOUNT_ID}/change_of_address_queue_${process.env.STAGE}_${process.env.CURRENT_COLOR}`,
+      });
+    });
+    await Promise.all(cmds.map(cmd => sqs.send(cmd)));
```
Perhaps we could do a batch here? A la `shared/admin-tools/aws/sqsHelper.ts::addToQueue`?

jtdevos replied:
I agree that should do batching where possible, my only reservation is doing so in the context of this v2->v3 upgrade task.  Let's give it a shot. If it works, great, and if it ends up breaking things I feel we should revert and punt the idea to a separate devex task.

jimlerza reviewed (commented):
Couple of thoughts at this stage:

- I think we should pursue your idea of a single client, like we did for OpenSearch.
- The email-related files modified in this PR will likely have conflicts with our SES branch. It might be easier to merge that branch into this one now and just make them depend on each other.

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jtdevos committed:
10337: fixed worker.test.ts

jtdevos committed:
Merge branch 'staging' into devex-10337-convert-sqs

jtdevos committed:
10337: merging devex-10338-convert-ses

jtdevos committed:
10337: leverage addtoQueue for batch messages

jtdevos committed:
reverting batch send for now (see notes)

Tried to leverage /shared/admin-tools/sqsHelper but it is
unavailable to code in /web-api. I don't want to muddy up a
simple v2->v3 upgrade with more extensive refactoring, so I think
I'll put it in a separate PR.

jtdevos committed:
10337: reverting getAwsClientConfig - may revisit when we refactor getMessagingClient()

jtdevos committed:
merged latest from devex-10338-convert-ses

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jtdevos committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
devex-10337: don't hard-code a region when instatiating the SQS client

jimlerza committed:
devex-10337: don't hard-code a region when instatiating the SES client

jimlerza committed:
devex-10337: use the region that is contextual to the running lambda when instantiating the SES client

jimlerza committed:
devex-10337: go back to hard-coding us-east-1, but only for SES

jimlerza committed:
devex-10337: remove unnecessary underscores

jimlerza committed:
devex-10337: typescript fixes for sendEmailEventToQueue

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs

jimlerza committed:
Merge branch 'staging' into devex-10337-convert-sqs
